File: backend/src/routes/notes.ts
import { Router } from 'express';
import { listNotes, allNotes, insertNote, updateNote, deleteNote } from '../services/repo.js';

export const notesRouter = Router();

notesRouter.get('/', (req, res) => {
  const target = req.query.target as string | undefined;
  if (!target) return res.json(allNotes());
  const targetId = req.query.targetId != null ? Number(req.query.targetId) : null;
  res.json(listNotes(target, targetId));
});

notesRouter.post('/', (req, res) => {
  const { target, targetId, body } = req.body ?? {};
  if (!target || !body) return res.status(400).json({ error: 'target and body required' });
  res.json(insertNote(target, targetId ?? null, body));
});

notesRouter.patch('/:id', (req, res) => {
  const { body } = req.body ?? {};
  const note = updateNote(Number(req.params.id), body ?? '');
  if (!note) return res.status(404).json({ error: 'Not found' });
  res.json(note);
});

notesRouter.delete('/:id', (req, res) => {
  deleteNote(Number(req.params.id));
  res.json({ ok: true });
});

File: backend/app/routers/notes.py
from __future__ import annotations

from fastapi import APIRouter, Request

from ..core.errors import ApiError
from ..services import repo

router = APIRouter()


@router.get("")
async def list_notes(target: str | None = None, targetId: int | None = None) -> list[dict]:
    if not target:
        return repo.all_notes()
    return repo.list_notes(target, targetId)


@router.post("")
async def add_note(request: Request) -> dict:
    body = await request.json() or {}
    if not body.get("target") or not body.get("body"):
        raise ApiError("target and body required", 400)
    return repo.insert_note(body["target"], body.get("targetId"), body["body"])


@router.patch("/{note_id}")
async def update_note(note_id: int, request: Request) -> dict:
    body = await request.json() or {}
    note = repo.update_note(note_id, body.get("body") or "")
    if not note:
        raise ApiError("Not found", 404)
    return note


@router.delete("/{note_id}")
async def delete_note(note_id: int) -> dict:
    repo.delete_note(note_id)
    return {"ok": True}
